lib/services/peer_info_service.dart

```diff
@@ -20,23 +20,62 @@ class PeerInfoService with ChangeCallbackProvider {
           .where((peerInfo) => peerInfo.id != null && peerInfo.name.isNotEmpty)
           .map((peerInfo) => MapEntry(peerInfo.id!, peerInfo.name)));
 
-  Future<void> upsert(PeerInfo peerInfo) async {
-    await _repository.upsert(peerInfo);
+  Future<void> upsert(
+    PeerInfo peerInfo, {
+    bool mergeWithExistent = true,
+  }) async {
+    if (mergeWithExistent && peerInfo.id != null) {
+      await _repository.runTransaction((txn) async {
+        final existentPeerInfo = await _repository.get(peerInfo.id!, txn: txn);
+        if (existentPeerInfo != null) {
+          existentPeerInfo.locations
+              .forEach((location) => peerInfo.addPeerLocation(location));
+
+          // if existent peer info is active and the public key is the same, the new peer info should also be active
+          if (existentPeerInfo.status == Status.active &&
```

Comment:
This doesn't matter, the updated one should aways have the status of the existentPeerInfo.
A updates B's publicKey => B still has the correct publicKey of A, no introductionMessage (key exchange) needed